mmengine/config/lazy.py

```diff
@@ -0,0 +1,225 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import importlib
+from typing import Any, Optional, Union
+
+from mmengine.utils import is_seq_of
+
+
+class LazyObject:
+    """LazyObject is used to lazy initialize the imported module during parsing
+    the configuration file.
+
+    During parsing process, the syntax like:
+
+    Examples:
+        >>> import torch.nn as nn
+        >>> from mmdet.models import RetinaNet
+        >>> import mmcls.models
+        >>> import mmcls.datasets
+        >>> import mmcls
+
+    Will be parsed as:
+
+    Examples:
+        >>> # import torch.nn as nn
+        >>> nn = lazyObject('torch.nn')
+        >>> # from mmdet.models import RetinaNet
+        >>> RetinaNet = lazyObject(['mmdet.models', 'RetinaNet'])
+        >>> # import mmcls.models; import mmcls.datasets; import mmcls
+        >>> mmcls = lazyObject(['mmcls', 'mmcls.datasets', 'mmcls.models'])
+
+    ``LazyObject`` records all module information and will be further
+    referenced by configuration file.
+
+    Args:
+        module (str or list or tuple): The module name to be imported.
+        imported (str, optional): The imported module name. Defaults to None.
+    """
+
+    def __init__(self,
+                 module: Union[str, list, tuple],
+                 imported: Optional[str] = None,
+                 location=None):
+        if (not isinstance(module, str) and not is_seq_of(module, str)):
+            raise TypeError('module should be `str`, `list`, `tuple` or '
+                            f'None, but got {type(module)}, this might be '
```

Comment:
Can the module be None?

Reply:
Both `from ... import xxx` and `import xxx` require that the `xxx` after the `import`. Therefore module could not be `None`